src/Components/ProductListing.js

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect} from "react";
+import axios from "axios";
+import { useDispatch } from "react-redux";
+import { setProducts } from "../redux/actions/productActions";
+import ProductComponent from "./ProductComponent";
+
+const ProductPage = () => {
+  const dispatch = useDispatch();
+  const fetchProducts = async () => {
+    const response = await axios
+      .get("https://api.spoonacular.com/recipes/random?apiKey=72c1f614304941ff8567b8d860bda7aa")

```

Comment:
Never expose the API URLs, API Keys or other sensitive information directly inside the code. Use environment configs.

- [ ] change